Lib/distutils/command/check.py

```diff
@@ -135,7 +135,7 @@ def _check_rst_data(self, data):
                           error_handler=settings.error_encoding_error_handler)
 
         document = nodes.document(settings, reporter, source=source_path)
-        document.note_source(source_path, -1)
+        document.note_source(StringIO(), -1)
```

Comment:
What does note_source do?

Reply:
it sets `current_source` and `current_line`, the former also being “Path to or description of the input source being processed”

will change

Reply:
Would it be easy to add a test for this too?

Reply:
so distutils went against [this recommendation](https://github.com/docutils-mirror/docutils/blob/e88c5fb08d5cdfa8b4ac1020dd6f7177778d5990/docutils/nodes.py#L1165-L1166) and instantiates the class directly.

[`new_document`](https://github.com/docutils-mirror/docutils/blob/e88c5fb08d5cdfa8b4ac1020dd6f7177778d5990/docutils/utils/__init__.py#L442-L443) does exactly the same thing except that it also creates a reporter, to i think we can skip a test here.

note that `decode_path` is just something like `ensure_unicode` and while `new_document` talks of paths, the `document` class says [“Path to or description of the input source being processed”](https://github.com/docutils-mirror/docutils/blob/e88c5fb08d5cdfa8b4ac1020dd6f7177778d5990/docutils/nodes.py#L1173)

I didn't expect the Spanish Inquisition!

Reply:
Ok but I would just like to be sure about what the change does.  Is the source_path used in the error message or warning text maybe?  Even without a unit test, does the command-line output make sense for a human reader?

Reply:
@flying-sheep ping

Reply:
sorry! i looked into it, and it needs to be a path after all if you use a relative path for a `include` or `csv_table` directive.